src/main/java/ru/skypro/homework/controller/AdsController.java

```diff
@@ -0,0 +1,86 @@
+package ru.skypro.homework.controller;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import ru.skypro.homework.dto.ads.Ad;
+import ru.skypro.homework.dto.ads.Ads;
+import ru.skypro.homework.dto.ads.CreateOrUpdateAd;
+import ru.skypro.homework.dto.ads.ExtendedAd;
+import ru.skypro.homework.service.AdsService;
+
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping("/ads")
+@RequiredArgsConstructor
+public class AdsController {
+
+    private final AdsService adsService;
+
+    @GetMapping()
+    public ResponseEntity<Ads> getAllAds(Authentication authentication){
+        Ads allAds = adsService.getAllAds(authentication);
+
+        return ResponseEntity.ok()
+                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                .body(allAds);
+    }
+
+    @PostMapping()
+    public ResponseEntity<Ad> addAd(Authentication authentication, @RequestBody @Valid CreateOrUpdateAd properties, @RequestBody MultipartFile image){
```

Comment:
```suggestion
    public ResponseEntity<Ad> addAd(Authentication authentication, @RequestBody @Valid CreateOrUpdateAd properties, @RequestBody MultipartFile image){
```
```suggestion
    public ResponseEntity<Ad> addAd(Authentication authentication, @RequestPart @Valid CreateOrUpdateAd properties, @RequestPart MultipartFile image){
```
аннотации такие должны быть, иначе не будет работать